server/src/test/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregatorTests.java

```diff
@@ -618,47 +592,6 @@ public void testAsSubAggregation() throws IOException {
         }, mappedFieldTypes);
     }
 
-    public void testCacheAggregation() throws IOException {
-        final Directory directory = newDirectory();
-        final RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory);
-        final int numDocs = 10;
-        for (int i = 0; i < numDocs; i++) {
-            indexWriter.addDocument(singleton(new NumericDocValuesField("number", i + 1)));
-        }
-        indexWriter.close();
-
-        final Directory unmappedDirectory = newDirectory();
-        final RandomIndexWriter unmappedIndexWriter = new RandomIndexWriter(random(), unmappedDirectory);
-        unmappedIndexWriter.close();
-
-        final IndexReader indexReader = DirectoryReader.open(directory);
-        final IndexReader unamappedIndexReader = DirectoryReader.open(unmappedDirectory);
-        final MultiReader multiReader = new MultiReader(indexReader, unamappedIndexReader);
-        final IndexSearcher indexSearcher = newSearcher(multiReader, true, true);
-
-        final MappedFieldType fieldType = new NumberFieldMapper.NumberFieldType("number", NumberFieldMapper.NumberType.INTEGER);
-        final CardinalityAggregationBuilder aggregationBuilder = new CardinalityAggregationBuilder("cardinality").field("number");
-
-        final AggregationContext context = createAggregationContext(indexSearcher, null, fieldType);
-        final CardinalityAggregator aggregator = createAggregator(aggregationBuilder, context);
-        aggregator.preCollection();
-        indexSearcher.search(new MatchAllDocsQuery(), aggregator.asCollector());
-        aggregator.postCollection();
-
-        final InternalCardinality cardinality = (InternalCardinality) aggregator.buildAggregation(0L);
-
-        assertEquals(10.0, cardinality.getValue(), 0);
-        assertEquals("cardinality", cardinality.getName());
-        assertTrue(AggregationInspectionHelper.hasValue(cardinality));
-
-        // Test that an aggregation not using a script does get cached
-        assertTrue(context.isCacheable());
```

Comment:
This assert is the only thing this test adds over the refactored test above.  I don't know where we should be testing that `SearchExecutionContext` is setting this correctly, but I don't think this is the right place. And having an aggregation context reference at this point is not good.  Let's discuss if this test adds enough value that we should re-create it somewhere closer to the code actually being tested.

Reply:
:+1:

I wonder if we should assert that all aggs are cacheable in test-case unless we have some kind of flag.

Reply:
That seems sensible.  I'll try that and see if anything breaks.

Reply:
So exactly two tests broke, both of which are sorting by a script so it makes sense they shouldn't be cached.  With #90149 merged, I am able to communicate that information to the right place in the test framework, with minimal intrusion on the rest of the code.  So I think we should be all set here.